src/reason_pprint_ast.ml

```diff
@@ -3303,11 +3313,17 @@ class printer  ()= object(self:'self)
                   | None -> (
                     (* Standard application *)
                     (*reset here only because [function,match,try,sequence] are lower priority*)
-                    List.concat [
-                      [SourceMap (e.pexp_loc, (self#expression2 e))];
-                      List.map self#reset#label_x_expression_param l;
-                      attributesAsList;
-                    ],
+                    let exp = match detectJSXComponent e.pexp_desc x.pexp_attributes with
```

Comment:
You may also want to detect if "l" here could be a valid JSX children (at least a list). Otherwise `Foo.createElement "" [@JSX]` will be formatted as `<Foo ="" />;`, which is incorrect.